### PR TITLE
expr,sql: support for cbrt

### PR DESCRIPTION
**Pull request by andrioni**
Add cbrt (cubic root) support for f32, f64 and decimal, following the same approach as used for sqrt. In this case, Postgres also doesn't have a specialized cbrt implementation for decimals and casts to f64, so doing the same here is probably good enough.

This was done mostly as an exercise to understand how functions are defined inside Materialize.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5763)

### CODE REVIEW

File: src/sql/src/func.rs [hunk 1]
```diff
@@ -1315,6 +1315,14 @@ lazy_static! {
                 params!(String) => UnaryFunc::TrimWhitespace, 885;
                 params!(String, String) => BinaryFunc::Trim, 884;
             },
+            "cbrt" => Scalar {
+                params!(Float32) => UnaryFunc::CbrtFloat32, oid::FUNC_CBRT_F32_OID;
+                params!(Float64) => UnaryFunc::CbrtFloat64, 1345;
```
Comment (andrioni) [inline src/sql/src/func.rs:1320]: This OID was taken from Postgres, via `select 'cbrt'::regproc::oid;`

**Comment by mjibson** [comment]
If we ever care about high precision, a decimal implementation exists: https://github.com/cockroachdb/apd/blob/d020e156310a8c3e542de08b8fbcd33938cd45af/context.go#L554

**Comment by benesch** [comment]
Thanks very much, @andrioni! This is awesome. Our function selection logic has changed quite a bit in the past few years, so I took the opportunity to clean some stuff up:

  * I removed `sqrt(f32)`, because that's not a thing in PostgreSQL. Instead that falls into the `sqrt(f64)` case via the implicit `f32` -> `f64` cast.
  * I removed `cbrt(f32)` by the same logic.
  * I _also_ removed `cbrt(numeric)`, which does not exist in PostgreSQL—unlike `sqrt(numeric)`, which does exist! Note that it's still possible to call `cbrt` on a `numeric` value because of the implicit `f64` -> `numeric` cast—it's just that you'll get a `double precision` out rather than a `numeric`.
 
Tests adjusted accordingly. I also added a release note and docs. I'll merge on green unless @philip-stoev has anything else he'd like to see tested here.

**Comment by andrioni** [comment]
Thanks for the quick review, @benesch! I imagine it's better then to try to keep the semantics aligned with Postgres in cases like this?

**Review by philip-stoev — COMMENTED** [review]
Thank you for the tests!

**Comment by benesch** [comment]
> Thanks for the quick review, @benesch! I imagine it's better then to try to keep the semantics aligned with Postgres in cases like this?

Yes, exactly! Over time we've leaned more and more into the strategy of just matching PostgreSQL's behavior exactly when possible.

**Pull request merged**